concordium-consensus/tests/scheduler/SchedulerTests/Helpers.hs

```diff
@@ -1,24 +1,430 @@
 
-simpleTransferCostWithMemo1 :: Word64 -> Energy
-simpleTransferCostWithMemo1 memoSize = Cost.baseCost (Types.transactionHeaderSize + 41 + 2 + memoSize) 1
+simpleTransferCostWithMemo1 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo1 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+
+simpleTransferCostWithMemo2 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo2 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+        + Cost.simpleTransferCost
+
+-- | Monad that implements the necessary constraints to be used for running the scheduler.
+newtype PersistentBSM pv a = PersistentBSM
+    { _runPersistentBSM ::
+        BS.PersistentBlockStateMonad
+            pv
+            (BS.PersistentBlockStateContext pv)
+            (Blob.BlobStoreM' (BS.PersistentBlockStateContext pv))
+            a
+    }
+    deriving
+        ( Applicative,
+          Functor,
+          Monad,
+          BS.AccountOperations,
+          BS.ContractStateOperations,
+          BS.ModuleQuery,
+          MonadProtocolVersion,
+          BlockStateTypes,
+          Blob.MonadBlobStore,
+          BS.BlockStateQuery,
+          BS.BlockStateOperations,
+          MonadIO,
+          MonadCache BS.ModuleCache,
+          BS.BlockStateStorage
+        )
+
+deriving instance
+    (Types.AccountVersionFor pv ~ av) =>
+    MonadCache (BS.AccountCache av) (PersistentBSM pv)
+
+instance MonadLogger (PersistentBSM pv) where
+    logEvent _ _ _ = return ()
+
+instance TimeMonad (PersistentBSM pv) where
+    currentTime = return $ read "1970-01-01 13:27:13.257285424 UTC"
+
+-- |Call a function for each protocol version, returning a list of results.
+-- This is used to run a test against every protocol version.
+forEveryProtocolVersion ::
+    (forall pv. (Types.IsProtocolVersion pv) => Types.SProtocolVersion pv -> String -> a) ->
+    [a]
+forEveryProtocolVersion check =
+    [ check Types.SP1 "P1",
+      check Types.SP2 "P2",
+      check Types.SP3 "P3",
+      check Types.SP4 "P4",
+      check Types.SP5 "P5"
+    ]
+
+-- |Construct a test block state containing the provided accounts.
+createTestBlockStateWithAccounts ::
+    (Types.IsProtocolVersion pv) =>
+    [Basic.Account (Types.AccountVersionFor pv)] ->
+    PersistentBSM pv (BS.HashedPersistentBlockState pv)
+createTestBlockStateWithAccounts accounts =
+    BS.initialPersistentState
+        (initialSeedState (Hash.hash "") 1_000)
+        DummyData.dummyCryptographicParameters
+        accounts
+        DummyData.dummyIdentityProviders
+        DummyData.dummyArs
+        DummyData.dummyKeyCollection
+        DummyData.dummyChainParameters
+
+-- |Run test block state computation provided a cache size.
+runTestBlockStateWithCacheSize :: Int -> PersistentBSM pv a -> IO a
+runTestBlockStateWithCacheSize cacheSize computation =
+    Blob.runBlobStoreTemp "." $
```

Comment:
Also, it might be better to change the PersistentBSM to use the MemBlobStore for tests.

Reply:
Although we can do that later as well.